weave

```diff
@@ -1027,6 +1031,17 @@ wait_for_status() {
     done
 }
 
+# Parse the output of `ip addr show`, outputing weave addresses of kind $1
+addrs_from_ip_addr_show() {
+    while read first second rest ; do
+        if [ "$first" != "${first%:}" ] ; then
+            iname="${second%:}"
+        elif [ "$first" = "$1" -a \( "$iname" = "weave" -o "$iname" != "${iname#ethwe}" \) ] ; then
+            echo $second
+        fi
+    done
+}
```

Comment:
Thanks; I didn't know about `-o`.

We don't need the MAC for internal purposes; could split into one implementation for `weave ps` and one for reclaims etc.?